c/tskit/trees.h

```diff
@@ -88,7 +88,9 @@ typedef struct {
     tsk_mutation_t *site_mutations_mem;
     tsk_mutation_t **site_mutations;
     tsk_size_t *site_mutations_length;
-    /* The underlying tables */
+    /** @brief  The table collection underlying this tree sequence, note that this
```

Comment:
Maybe strengthen that assertion up a bit:
``suggestion
    /** @brief  The table collection underlying this tree sequence. This table collection must be treated as read-only, and any changes to it will lead to undefined behaviour.
```